stdlib/Profile/test/allocs.jl

```diff
@@ -0,0 +1,57 @@
+using Test
+using Profile: Allocs
+
+@testset "alloc profiler doesn't segfault" begin
+    res = Allocs.@profile sample_rate=1.0 begin
+        # test the allocations during compilation
+        using Base64
+    end
+    profile = Allocs.fetch()
+
+    @test length(profile.allocs) > 0
+    first_alloc = profile.allocs[1]
+    @test first_alloc.size > 0
+    @test length(first_alloc.stacktrace) > 0
+    @test length(string(first_alloc.type)) > 0
+end
+
+@testset "alloc profiler works when there are multiple tasks on multiple threads" begin
+    NUM_TASKS = 1000
+
+    # TODO: is this always true in CI?
+    @test Threads.nthreads() > 1
+
+    function do_work()
+        ch = Channel{Vector{Int}}(Inf)
+        @sync for i in 1:NUM_TASKS
+            Threads.@spawn begin
+                # generate garbage
+                put!(ch, zeros(100))
+            end
+        end
+        close(ch)
+    end
+
+    # call once to make sure it's compiled
+    do_work()
+
+    res = Allocs.@profile sample_rate=1 begin
+        do_work()
+    end
+    profile = Allocs.fetch()
+
+    # expecting at least 3 allocations per task:
+    # 1. the task
+    # 2. the vector
+    # 3. the buffer inside the vector
+    @test length(profile.allocs) >= 3*NUM_TASKS
+    println(length(profile.allocs))
+    first_alloc = profile.allocs[1]
+    @test first_alloc.size > 0
+    @test length(first_alloc.stacktrace) > 0
+    @test length(string(first_alloc.type)) > 0
+
```

Comment:
I guess we could actually test for the expected types in here too? Something like
```suggestion
    # expecting at least 3 allocations per task:
    # 1. the task
    # 2. the vector
    # 3. the buffer inside the vector
    @test length(profile.allocs) >= 3*NUM_TASKS
    println(length(profile.allocs))
    first_alloc = profile.allocs[1]
    @test first_alloc.size > 0
    @test length(first_alloc.stacktrace) > 0
    @test length(string(first_alloc.type)) > 0
    @testset for type in (Task, Vector, #= buffer =#)
        @test length(filter(a->a.type <: type, profile.allocs)) >= NUM_TASKS
    end
```
Or something like that? (that passed for me when i tried it locally.)

Reply:
Done.